### PR TITLE
addition of contributing.md

### PR DESCRIPTION
**Pull request by derekwolpert**
Contributing.md covers:
- context to ADX, Rearc and why we are contributing to the ADX platform
- overview of tools and directory layout of our ADX repos
- guidance on how to contact us if there is an issue (GitHub issue, email, etc.)
- guidance on opening a pull request on one of our ADX projects

Very open to making further adjustments or adding additional info, let me know

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Contributing To A Rearc AWS Data Exchange Product
```
Comment (milin82) [inline CONTRIBUTING.md:1]: suggestion: `Contributing to a Rearc AWS Data Exchange product`

Comment (derekwolpert) [reply CONTRIBUTING.md]: I've adjusted the title per your suggestion, and made a few other minor edits

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# Contributing to a Rearc AWS Data Exchange product
+
+🎉🥳 First of all, **THANK YOU** for being interested in contributing to one of our projects 🎉🥳
+
+From the start of our AWS Data Exchange (ADX) initiative we knew we wanted to have an open dialogue with our subscribers. While we try our best to optimize the packaging of our ADX products to be as accommodating as possible, we appreciate there may be use-cases we have overlooked. By contributing to this project, you have an opportunity to improve the usability of the data included with this product not only for yourself, but for anyone who subscribes to this product through ADX. If your contribution is applicable to more than one of our ADX listings, we will gladly improve those other products as well!
```
Comment (mwkaufman) [inline CONTRIBUTING.md:5]: Overall I think this Markdown is a bit to get through. I'd consider this paragraph for cutting.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# Contributing to a Rearc AWS Data Exchange product
+
+🎉🥳 First of all, **THANK YOU** for being interested in contributing to one of our projects 🎉🥳
```
Comment (mwkaufman) [inline CONTRIBUTING.md:3]: Good start.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+From the start of our AWS Data Exchange (ADX) initiative we knew we wanted to have an open dialogue with our subscribers. While we try our best to optimize the packaging of our ADX products to be as accommodating as possible, we appreciate there may be use-cases we have overlooked. By contributing to this project, you have an opportunity to improve the usability of the data included with this product not only for yourself, but for anyone who subscribes to this product through ADX. If your contribution is applicable to more than one of our ADX listings, we will gladly improve those other products as well!
+
+You can learn about Rearc’s data services by visiting [https://www.rearc.io/data]( https://www.rearc.io/data).
```
Comment (mwkaufman) [inline CONTRIBUTING.md:7]: I don't think we need to drive people in our CONTRIBUTORS.md to that (mostly marketing at this point) website.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+Below is a set of guidelines for contributing to any of the ADX product repositories available on the [Rearc Data GitHub account](https://github.com/rearc-data).
```
Comment (mwkaufman) [inline CONTRIBUTING.md:9]: I'd consider cutting.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  * [What tools are you using throughout your ADX Products?](#what-tools-are-you-using-throughout-your-adx-products)
+  * [Directory Layout](#directory-layout)
+- [How can I contribute?](#how-can-i-contribute)
+  * [Report an Issue/Bug](#report-an-issuebug)
+  * [Submit an Improvement/Suggestion](#submit-an-improvementsuggestion)
+  * [Pull Request](#pull-request)
+- [Additional Resources](#additional-resources)
+
+## I have a question and/or issue to report. What is the best way to contact Rearc?
+If you have an issue to report specific to the source code in this repository you can [open a GitHub issue](https://github.com/rearc-data/publish-a-data-product-on-aws-data-exchange/issues). Before opening an issue please review the existing suggestions to see if your idea is already there. If already present, please comment on the existing issue instead of making a new one.
```
Comment (mwkaufman) [inline CONTRIBUTING.md:27]: Looks good but including a link might make it hard to sync this file between all our rearc-data repositories.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+#### What is Rearc's philosophy towards dataset formats?
+We try as much as possible to preserve the integrity of data we provide through ADX, and most of the time this means delivering datasets exactly as they were presented from their source. Sometimes we make minor alterations to datasets to provide wider usability for ADX subscribers (e.g. adjusting CSV files for SQL column naming conventions). For situations where we are unable to maintain the original data file format (often due to a request from the data consumer or a discrepancy in a source's delivery pipeline), we try to limit the extent of needed transformations as much as possible.
+
+#### What tools are you using throughout your ADX Products?
+- On our local devices we have chosen to use a combination of [Python 3](https://www.python.org), [Pip](https://pypi.org/project/pip/), [JQ](https://stedolan.github.io/jq/), [AWS CLI V2](https://aws.amazon.com/cli/) and other related developer tools. The specific tools utilized varies on a project-by-project basis. The Python portions of our source code can be adapted into another language with a supported AWS Lambda runtime.
```
Comment (mwkaufman) [inline CONTRIBUTING.md:46]: I would cut the last line about AWS supporting languages other than Python.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+- We are creating various AWS resources throughout the process of deploying and maintaining our ADX projects, including [Data Exchange](https://docs.aws.amazon.com/data-exchange/), [S3](https://docs.aws.amazon.com/s3/), [Lambda](https://docs.aws.amazon.com/lambda/) and [EventBridge](https://docs.aws.amazon.com/eventbridge/). The creation of most of the needed resources is streamlined by the [CloudFormation](https://docs.aws.amazon.com/cloudformation/) template included with this project, ([pre-processing-cfn.yaml](./pre-processing/pre-processing-cfn.yaml)).
```
Comment (mwkaufman) [inline CONTRIBUTING.md:48]: I think this whole section can be cut down to we use Python 3, CloudFormation, and Lambda roughly.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+│   │   └── source_data.py          # Code to acquire dataset and perform necessary data transformation
+│   └── pre-processing-cfn.yaml     # CloudFormation template to setup data provider automation
+├── dataset-description.md          # Dataset description that goes on the ADX listing
+├── product-description.md          # Product description that goes on the ADX listing
+├── init.sh                         # Initialization script to kick-off the automation
+├── README.md
+└── .gitignore
+```
+For more details on the technologies and directory layout used in our ADX products, please visit [Getting started with publishing a data product on AWS Data Exchange](https://github.com/rearc-data/publish-a-data-product-on-aws-data-exchange).
+
```
Comment (mwkaufman) [inline CONTRIBUTING.md:66]: I think linking to this repository is fine, but maybe the directory layout should live there.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+## How can I contribute?
+
+#### Report an Issue/Bug
+If you are experiencing an issue with the ADX product featured in this repository, the best way to contact us would be through [opening a GitHub issue](https://github.com/rearc-data/publish-a-data-product-on-aws-data-exchange/issues) in this repository.
+
+When opening an issue please **be as descriptive as possible**. If relevant please **provide information regarding your use-case, development configuration and environment**. The more specific you can be the easier it will be for us to identify and address the situation.
+
+#### Submit an Improvement/Suggestion
+If you have a suggestion for improving the ADX product included in this repository you are welcomed to [open a GitHub issue](https://github.com/rearc-data/publish-a-data-product-on-aws-data-exchange/issues) in this repository. While we cannot guarantee every suggestion will be implemented into our products, we will get back to you either way.
```
Comment (mwkaufman) [inline CONTRIBUTING.md:75]: So this is the third time we've asked them to open an issue, I think we should mostly have one paragraph prompting them to open an issue and one prompting them to e-mail us and try not to repeat ourselves.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+If you have any general feedback or suggestions for our ADX products, please contact us at data@rearc.io.
```
Comment (mwkaufman) [inline CONTRIBUTING.md:77]: Same idea.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+#### Pull Request
+One of the major reasons we have made our ADX project repositories public is because we want to better maintain and improve the quality of our data products. We believe our subscribers (you) can play a critical role in this process, and we actively encourage you to fork, branch and open a pull request on this repository.
```
Comment (mwkaufman) [inline CONTRIBUTING.md:80]: Repeats some of the stuff above.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+
+If you have any general feedback or suggestions for our ADX products, please contact us at data@rearc.io.
+
+#### Pull Request
+One of the major reasons we have made our ADX project repositories public is because we want to better maintain and improve the quality of our data products. We believe our subscribers (you) can play a critical role in this process, and we actively encourage you to fork, branch and open a pull request on this repository. 
+
+Before opening a pull request please familiarize yourself with the [tools](#what-tools-are-you-using-throughout-your-adx-products) and [directory layout](#directory-layout) used in this repository. If you are looking to improve the project's included datasets you should direct yourself to the [`pre-processing/pre-processing-code`](./pre-processing/pre-processing-code) folder, as this is where the gathering and transforming of data occurs.
```
Comment (mwkaufman) [inline CONTRIBUTING.md:82]: Repeats some of the stuff above.

**Review by mwkaufman — CHANGES_REQUESTED** [review]
I think we need to make this more concise.

File: CONTRIBUTING.md (hunk 2 above)
Comment (derekwolpert) [reply CONTRIBUTING.md]: Removed the paragraph that was on line 5

File: CONTRIBUTING.md (hunk 3 above)
Comment (derekwolpert) [reply CONTRIBUTING.md:3]: 🎉🥳

File: CONTRIBUTING.md (hunk 4 above)
Comment (derekwolpert) [reply CONTRIBUTING.md]: Removed sentence in line 7 driving users to the Rearc Data website, left it under "Additional Resources" at the bottom of the document

File: CONTRIBUTING.md (hunk 5 above)
Comment (derekwolpert) [reply CONTRIBUTING.md]: Removed sentence on line 9

File: CONTRIBUTING.md (hunk 6 above)
Comment (derekwolpert) [reply CONTRIBUTING.md]: Removed the section on line 26 and 27, and incorporated parts of it into the "Report an Issue/Bug or Submit an Improvement/Suggestion" section further down in the document

File: CONTRIBUTING.md (hunk 7 above)
Comment (derekwolpert) [reply CONTRIBUTING.md]: Removed the line about supporting languages other than python

File: CONTRIBUTING.md (hunk 8 above)
Comment (derekwolpert) [reply CONTRIBUTING.md]: Cut down "What tools are you using throughout your ADX Products?" section to just Python 3, CloudFormation, and Lambda. Followed up with a brief bullet stating that the exact tools used vary depending on the specific dataset being published

File: CONTRIBUTING.md (hunk 9 above)
Comment (derekwolpert) [reply CONTRIBUTING.md:36]: Completely removed directory layout section. Edited line 65 to be shorter, and placed it at the end of the "What tools are you using throughout your ADX Products?" section.

File: CONTRIBUTING.md (hunk 10 above)
Comment (derekwolpert) [reply CONTRIBUTING.md]: Not the third time anymore! Now it's the first and only! Combined the "Report an Issue/Bug" and "Submit an Improvement/Suggestion" sections together.

File: CONTRIBUTING.md (hunk 11 above)
Comment (derekwolpert) [reply CONTRIBUTING.md]: "Report an Issue/Bug" and "Submit an Improvement/Suggestion" sections have been combined to limit redundant/repeated details

File: CONTRIBUTING.md (hunk 12 above)
Comment (derekwolpert) [reply CONTRIBUTING.md]: It's no longer repeating prior details, and it's been edited for increased brevity

File: CONTRIBUTING.md (hunk 13 above)
Comment (derekwolpert) [reply CONTRIBUTING.md]: Lines 80 and 82 have been edited and combined.

**Review by mwkaufman — APPROVED** [review]
Much improved, let's discuss in the meeting before pushing it out everywhere.

**Review by milin82 — APPROVED** [review]

**Pull request merged**